### PR TITLE
CLOUD-2529 | Extend aws-cdk-github-actions to support private Github Repos and Custom Tokens

### PR DESCRIPTION
**Pull request by julienbonastre**

### CODE REVIEW

File: entrypoint.sh [hunk 1]
```diff
@@ -10,6 +10,12 @@ function parseInputs(){
 	fi
 }
 
+function configurePrivateGitPath(){
+  if [ -n "${INPUT_GITHUB_TOKEN}" ] && [ -n "${INPUT_GITHUB_PRIVATE_PATH}" ]; then
```
Comment (hiruna) [inline entrypoint.sh:14]: `-n` flag ahhh

File: entrypoint.sh [hunk 2]
```diff
@@ -10,6 +10,12 @@ function parseInputs(){
...
+    git config --global url."https://${INPUT_GITHUB_TOKEN}:x-oauth-basic@${INPUT_GITHUB_PRIVATE_PATH}".insteadOf "https://${INPUT_GITHUB_PRIVATE_PATH}"
```
Comment (hiruna) [inline entrypoint.sh:15]: 👌

**Review by hiruna — APPROVED** [review]
LGTM 👌

**Comment by julienbonastre** [comment]
Thanks @Hiruna 🤗🦦

**Pull request merged**